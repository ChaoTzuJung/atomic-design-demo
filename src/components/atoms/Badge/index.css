/* color */
.secondary-normal {
	color: white;
	background-color: var(--secondary);
}

.light-normal {
	color: #283e59;
	background-color: var(--light);
}

.green-soft {
	color: var(--green);
	background-color: #ccf7e5;
}

.red-soft {
	color: var(--red);
	background-color: #fad7dd;
}

.yellow-soft {
	color: var(--yellow);
	background-color: #fdf3d9;
}

.blue-soft {
	color: var(--primary);
	background-color: #d5e5fa;
}

.gray-soft {
	color: var(--secondary);
	background-color: #e2e6ed;
}

.cyan-soft {
	color: var(--cyan);
	background-color: #d7eff6;
}

.badge {
	display: inline-flex;
	padding: 0 4px;
	align-items: center;
	height: 20px;
	line-height: 1;
	font-size: 12px;
	font-weight: 500;
	white-space: nowrap;
	letter-spacing: 0.5px;
	border-radius: 3px;

	& > .text {
		transform: scale(0.85);
	}

	&.outline {
		color: var(--grayDark);
		border: 1px solid var(--grayDark);
		background-color: transparent;
	}
}
