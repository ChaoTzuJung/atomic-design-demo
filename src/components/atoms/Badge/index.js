import React from 'react';
import classnames from 'classnames';
import styles from './index.css';

const Badge = ({ text, colorType, className, outline }) => (
	<span className={classnames(styles.badge, className, styles[colorType], outline ? styles.outline : '')}>
		<div className={styles.text}>{text}</div>
	</span>
);

export default Badge;
