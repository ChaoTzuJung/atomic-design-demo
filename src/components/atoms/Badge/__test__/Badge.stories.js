import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs/react';

import Badge from 'components/atoms/Badge';

const stories = storiesOf('Commons|atoms/Badge', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => (
	<Badge
		colorType={select(
			'colorType',
			[
				'secondary-normal',
				'light-normal',
				'green-soft',
				'red-soft',
				'yellow-soft',
				'blue-soft',
				'gray-soft',
				'cyan-soft',
			],
			'secondary-normal',
		)}
		text={text('文字', 'label')}
		outline={boolean('isOutline')}
	/>
));
