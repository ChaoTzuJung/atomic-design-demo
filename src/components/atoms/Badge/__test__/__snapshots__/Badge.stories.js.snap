// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<Badge
  colorType="secondary-normal"
  text="label"
>
  <span
    className="badge secondary-normal"
  >
    <div
      className="text"
    >
      label
    </div>
  </span>
</Badge>
`;
